synapse/eventbus.py

```diff
@@ -378,14 +416,15 @@ class BusRef(EventBus):
     '''
     An object for managing multiple EventBus instances.
     '''
-    def __init__(self):
+    def __init__(self, ctor=None):
         EventBus.__init__(self)
+        self.ctor = ctor
+        self.lock = threading.Lock()
         self.ebus_by_name = {}
         self.onfini(self._onBusRefFini)
 
     def _onBusRefFini(self):
-        todo = list(self.ebus_by_name.values())
-        [ebus.fini() for ebus in todo]
+        [ebus.fini() for ebus in self.vals()]
```

Comment:
self.vals isn't  a member off busref or eventbus

Reply:
Looks like it is, at: https://github.com/vertexproject/synapse/pull/480/files#diff-8c27f6a6100452315bb1120577a5bab3R501

Reply:
hah i was looking for a defined object somewhere; not a method. my bad.